File: ios/scm/Models/Processor.swift
//
//  Processor.swift
//  scm
//
//  Created by Billy Anthony Bingtoyo on 17/10/24.
//

struct Processor: Hashable, Codable {
    var id: String
    var cluster_id: String
    var address: ProcessorAddress
    var name: String
    var version: String
}

struct ProcessorAddress: Hashable, Codable {
    var host: [UInt8]
    var port: UInt16
}

struct ProcessorRequest: Encodable, Equatable {
    var id: String
    var cluster_id: String
    var address: ProcessorAddress
    var name: String
}

struct ProcessorRef: Hashable, Codable {
    var id: String
    var name: String
}

struct ViewProcessor: Hashable, Codable {
    var id: String
    var cluster: CameraRef
    var camera: [ViewCamera]
    var address: ProcessorAddress
    var name: String
    var violation_count: Int
    var notification_count: Int
}

File: processor-web/app/types/processor.ts
export type Processor = {
  id: string;
  name: string;
  model: string;
  address: ProcessorAddress;
  webhook?: ProcessorWebhook;
  version: number;
};

export type ProcessorAddress = {
  host: [number, number, number, number];
  port: number;
};

export type ProcessorWebhook = {
  host: ProcessorWebhookHost;
  port?: number;
  path: {
    evidence: string;
    update: string;
  };
  secure: boolean;
};

export type ProcessorWebhookHost = string | [number, number, number, number];
